backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/ELKDiagramConverter.java

```diff
@@ -78,15 +78,31 @@ public ELKDiagramConverter(TextBoundsService textBoundsService, ImageSizeProvide
         this.imageNodeStyleSizeProvider = new ImageNodeStyleSizeProvider(this.imageSizeProvider);
     }
 
+    public ELKConvertedDiagram convert(Diagram diagram, Node parentNode) {
+        List<Node> childNodes = this.initializeNodes(List.of(parentNode));
+
+        Diagram initializedDiagram = Diagram.newDiagram(diagram)
+                .nodes(childNodes)
+                .build();
```

Comment:
I don't think I understand the necessity of this method but if I understand your code correctly, let's say that I have a diagram with 10 nodes. You will initialize one node and then create a new diagram by replacing the list of 10 nodes with the initialized node. The end result of this function does not match at all the given diagram. This diagram can have edges with a sourceId which do not match anything anymore.